modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/ddl/DdlCommandHandler.java

```diff
@@ -0,0 +1,281 @@
+
+            if (idxInfo.getSecond()) {
+                idx0.desc();
+            }
+
+            idx0.done();
+        }
+
+        String fullName = TableDefinitionImpl.canonicalName(cmd.schemaName(), cmd.tableName());
+
+        tableManager.alterTable(fullName, chng -> chng.changeIndices(idxes -> {
+            if (idxes.get(cmd.indexName()) != null) {
+                throw new IndexAlreadyExistsException(cmd.indexName());
+            }
+
+            idxes.create(cmd.indexName(), tableIndexChange -> convert(idx.build(), tableIndexChange));
+        }));
+    }
+
+    /** Handles create index command. */
+    private void handleDropIndex(DropIndexCommand cmd) {
+        String fullName = TableDefinitionImpl.canonicalName(cmd.schemaName(), cmd.tableName());
+
+        tableManager.alterTable(fullName, chng -> chng.changeIndices(idxes -> {
+            if (idxes.get(cmd.indexName()) == null) {
+                if (!cmd.ifExist()) {
+                    throw new IndexAlreadyExistsException(cmd.indexName());
+                }
+            }
+
+            idxes.delete(cmd.indexName());
+        }));
+    }
+
+    /**
+     * Adds a column according to the column definition.
+     *
+     * @param fullName Table with schema name.
+     * @param colsDef  Columns defenitions.
+     * @param colNotExist Flag indicates exceptionally behavior in case of already existing column.
+     */
+    private void addColumnInternal(String fullName, Set<ColumnDefinition> colsDef, boolean colNotExist) {
+        tableManager.alterTable(
+                fullName,
+                chng -> chng.changeColumns(cols -> {
+                    Map<String, String> colNamesToOrders = columnOrdersToNames(chng.columns());
+
+                    Set<ColumnDefinition> colsDef0;
+
+                    if (!colNotExist) {
+                        colsDef.stream().filter(k -> colNamesToOrders.containsKey(k.name())).findAny()
+                                .ifPresent(c -> {
+                                    throw new ColumnAlreadyExistsException(c.name());
+                                });
+
+                        colsDef0 = colsDef;
+                    } else {
+                        colsDef0 = colsDef.stream().filter(k -> !colNamesToOrders.containsKey(k.name())).collect(Collectors.toSet());
+                    }
+
+                    for (ColumnDefinition colDef : colsDef0) {
+                        cols.create(colDef.name(), colChg -> convert(colDef, colChg));
+                    }
+                }));
+    }
+
+    /**
+     * Adds a column according to the column definition.
+     *
+     * @param fullName Table with schema name.
+     * @param colNames  Columns defenitions.
+     * @param colNotExist Flag indicates exceptionally behavior in case of already existing column.
+     */
+    private void dropColumnInternal(String fullName, Set<String> colNames, boolean colExist) {
+        tableManager.alterTable(
```

Comment:
does this call respect `IF EXISTS` flag? I mean in the case `ALTER TABLE IF EXISTS ...`

Reply:
ok, catch it, append comment, this need to be implementad under : IGNITE-15926

Reply:
In IGNITE-15926 "if exists" is related to table, that means "alter table" operation will not be applied and return False if the table is not exists.
Here, "colExists" relates to column... but it is not clear for which one when a Set<> is provided?

Reply:
BTW, "IF EXISTS" construction is not part of SQL standard (92, 2003 or 2011). So, we could treat it as we want. 
Imho, supporting "if exists" related to the table - is must have feature, but supporting it on per-column basis is overkill.

Reply:
you are right, we discuss it already and found that it usefull.